### PR TITLE
Define and use some types

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR is a prep PR for adding edmf. It makes more sense to use dispatch over further specializing with `Val` over more symbols that come with using edmf. So, this PR adds and uses some types, some of which will be used to create an edmf model, so that we can create the state vector. I think that the cache is nearly finished / compatible, so we're getting pretty close.

None of this is final, I think there's still many changes to come, it's just a small step towards being compatible with edmf requirements.

Once all of this is in, @dennisYatunin can apply the driver revamp and I suspect much of this will be more polished

### CODE REVIEW

**Review by dennisYatunin — APPROVED** [review]

File: examples/hybrid/define_post_processing.jl
Comment (jiahe23) [inline examples/hybrid/define_post_processing.jl:135]: Can we call them `PotentialTemperature`, `TotalEnergy` instead of `ρθForm`, `ρeForm`

Comment (jiahe23) [reply examples/hybrid/define_post_processing.jl:135]: And `EquilMoistModel`

Comment (dennisYatunin) [reply examples/hybrid/define_post_processing.jl:135]: And maybe `EnergyVariable` instead of `EnergyForm`?

File: examples/hybrid/types.jl
Comment (jiahe23) [inline examples/hybrid/types.jl:4]: How about `EquilMoistModel`, `NonEquilMoistModel`?

File: examples/hybrid/types.jl
Comment (jiahe23) [inline examples/hybrid/types.jl:21]: I'd prefer `PotentialTemperature`, `TotalEnergy`, `InternalEnergy`

File: examples/hybrid/define_post_processing.jl
Comment (charleskawczynski) [reply examples/hybrid/define_post_processing.jl:135]: Sure, no problem. I kind of like the `ρθForm` and `ρeForm`, just because `PotentialTemperature` could cause confusion with `θ` itself. But I don't have strong opinions on names, here.

File: examples/hybrid/types.jl
Comment (charleskawczynski) [reply examples/hybrid/types.jl:4]: Definitely better, thanks!

**Comment by jiahe23** [comment]
I am working on have #438 merged in soon. Once done, the held_suarez paper plots will need changes accordingly.

**Comment by charleskawczynski** [comment]
> I am working on have #438 merged in soon. Once done, the held_suarez paper plots will need changes accordingly.

Sounds good, I'll rebase after #438 merges, I think I'd like to merge #445 before this.

**Comment by charleskawczynski** [comment]
Waiting for #447 to merge since I'm getting a formatter error..

File: examples/hybrid/define_post_processing.jl
Comment (charleskawczynski) [reply examples/hybrid/define_post_processing.jl]: I went with @jiahe23's initial suggestion+ `InternalEnergy`

**Pull request merged**
